### DISCUSSION TITLE
Questions about localization

### DISCUSSION THREAD

**Discussion by cary-rowen** [edrlab/thorium-reader#1663] [Q&A] [answered]
Hi @llemeurfr @panaC 
Thanks for your reply in #1371  I still seem a little confused, I have the following questions:

1. I found that "thorium-reader/src/resources/locales/zh-cn.json" contains many empty strings, do I need to start from this file if I want to contribute translations.
2. If thorium-reader adds new translatable strings in the new version, how should I update it to Simplified Chinese, Need to run ```npm run i18n-check``` to get new translatable strings?
3. I see that the maintainer of early simplified Chinese is @maitreya2019, I wonder if @maitreya2019 is still contributing translations.
4. I found that "thorium-reader/src/resources/information/zh-cn.xhtml" is already translated in Chinese, but it looks more like machine translation.
5. What should I do if I want to test my translation after translating, do I need to create a thorium-reader compilation environment locally?

My understanding: For Simplified Chinese I only need to translate the two files mentioned in (1) and (4), the former is the UI and the latter is the introduction information of the project, is that right?

Grateful

Cary

**Comment by danielweck** [comment]
Follow-up from issue: https://github.com/edrlab/thorium-reader/pull/1371#issuecomment-1086611358

**Comment by danielweck** [comment]
> 1. I found that "thorium-reader/src/resources/locales/zh-cn.json" contains many empty strings, do I need to start from this file if I want to contribute translations.

Hello, thank you very much for offering to help maintain the translation. Yes, the empty strings are intentionally placed to indicate that there are missing translated strings:
https://github.com/edrlab/thorium-reader/blob/develop/src/resources/locales/zh-cn.json

**Comment by danielweck** [comment]
> 2\. If thorium-reader adds new translatable strings in the new version, how should I update it to Simplified Chinese, Need to run `npm run i18n-check` to get new translatable strings?

You do not have to worry about calling the localization developer tools, we take care of it whenever we add an English string (so that other languages are updated with the empty strings `""`.

https://github.com/edrlab/thorium-reader/wiki/Localization-(l10n)-language-translations#tooling

https://github.com/edrlab/thorium-reader/blob/2a835e94b84f7b15f69b85676fbb9e4869e6b438/package.json#L71-L74

**Comment by danielweck** [comment]
> 3\. I see that the maintainer of early simplified Chinese is @maitreya2019, I wonder if @maitreya2019 is still contributing translations.

Feel free to contact them to coordinate your efforts :)
We are keeping track of contributions in this wiki page:
https://github.com/edrlab/thorium-reader/wiki/Localization-(l10n)-language-translations#chinese

**Comment by danielweck** [comment]
> 4\. I found that "thorium-reader/src/resources/information/zh-cn.xhtml" is already translated in Chinese, but it looks more like machine translation.

We welcome your help to correct errors in the HTML:

https://github.com/edrlab/thorium-reader/blob/develop/src/resources/information/zh-cn.xhtml

**Comment by danielweck** [comment]
> 5\. What should I do if I want to test my translation after translating, do I need to create a thorium-reader compilation environment locally?

Ideally, yes. Instructions:

* Install NodeJS 16.14.2 https://nodejs.org
* You can install the GitHub GUI app ( https://desktop.github.com ), or use the "git" command line (recommended):
* `git clone https://github.com/edrlab/thorium-reader.git` (this creates a folder named `thorium-reader`)
* `cd thorium-reader`
* `npm i`
* `npm run start:dev:quick`

That's it :)

**Comment by danielweck** [comment, marked as answer]
> My understanding: For Simplified Chinese I only need to translate the two files mentioned in (1) and (4), the former is the UI and the latter is the introduction information of the project, is that right?

* JSON translate:  https://github.com/edrlab/thorium-reader/blob/develop/src/resources/locales/zh-cn.json
* HTML translate: https://github.com/edrlab/thorium-reader/blob/develop/src/resources/information/zh-cn.xhtml

Thank you very much :) 👍

**Comment by cary-rowen** [comment]
Hi @danielweck 
Thank you very much for your answer, I think I got it.

grateful

**Comment by cary-rowen** [comment]
I seem to get an error when I run ```npm i```, the log is as follows:
[2022-04-06T09_28_05_544Z-debug-0.log](https://github.com/edrlab/thorium-reader/files/8425595/2022-04-06T09_28_05_544Z-debug-0.log)

**Reply by danielweck** [reply]
Ah sorry, this is the Divina Github NPM dependency, failing integrity check because of NPM command line tool strict checks. The solution is:
* `node scripts/package-lock-patch.js`
* ...then `npm i`

**Reply by danielweck** [reply]
README documentation (I forgot about this step ... I am on MacOS where this problem doesn't seem to occur):
https://github.com/edrlab/thorium-reader/blob/develop/README.md#install-dependencies

**Comment by cary-rowen** [comment]
Hi @danielweck 

Thanks for the further answer, it's really great, it solved my problem perfectly, I'll keep reading the readme to know more.

Grateful
